File: algorithms/implementation/p09_caesar_cipher/caesar_cipher.java
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int main(){
    int n;
    cin >> n;
    string s;
    cin >> s;
    int k;
    cin >> k;
    
    for (auto &c : s) {
        if (isalpha(c)) {
            int base = isupper(c) ? (int)'A' : (int)'a';
            cout << (char)(base + ((int)c - base + k) % 26);
        } else {
            cout << c;
        }
    }
    cout << endl;
    return 0;
}

File: algorithms/implementation/p09_caesar_cipher/caesar_cipher.py
#!/bin/python3

import sys

n = int(input().strip())
s = input().strip()
k = int(input().strip())

split = list(s)
for i in range(n):

    if not split[i].isalpha():
        print(split[i], end="")
    else:
        base = 'A' if split[i].isupper() else 'a'
        val = ord(base) + (ord(split[i]) - ord(base) + k) % 26
        print(chr(val), end="")

print()
